### DISCUSSION TITLE
esbuild

### DISCUSSION THREAD

**Discussion by j4k0xb** [nicoespeon/abracadabra#959] [Show and tell]
https://github.com/nicoespeon/abracadabra/blob/b10f3b5159ead7dcf1397686f91bdab618e27168/docs/adr/0011-use-babel-jest-instead-of-ts-jest.md?plain=1#L23

Imo esbuild is now mature enough to be considered: Both node and web extension bundles work, there are source maps and a [bundle size analyzer](https://esbuild.github.io/analyze/).
It results in a much more pleasant development experience when pressing F5: **0.6s** build time instead of 73s.
Are you interested in a PR? https://github.com/j4k0xb/abracadabra/tree/esbuild

Other optimization ideas:
https://github.com/nicoespeon/abracadabra/blob/b10f3b5159ead7dcf1397686f91bdab618e27168/package.json#L49
currently takes 26s. It could also use esbuild or explore if running [vitest](https://vitest.dev/) directly on the source files is faster (if it isn't too much work)

**Comment by nicoespeon** [comment]
@j4k0xb that sounds super promising and interesting. I will have some time available to work on Abracadabra around Christmas, so feel free to open a PR. Always happy to optimize the dev or user experience for the extension 😄

**Comment by nicoespeon** [comment]
Implemented with #960, thank you again!
